earth2mip/initial_conditions/gfs.py

```diff
@@ -198,15 +199,18 @@ def get(
             + "(needs to be past 10 days)"
         )
 
-    # Make temp grib folder
-    pathlib.Path(GFS_CACHE).mkdir(parents=True, exist_ok=True)
-    # Get index file
-    gfs_chunks = get_gfs_chunks(time_gfs)
+    if download_gfs:
+        # Make temp grib folder
+        pathlib.Path(GFS_CACHE).mkdir(parents=True, exist_ok=True)
+        # Get index file
+        gfs_chunks = get_gfs_chunks(time_gfs)
 
-    # Loop through channels and download grib of each
-    logger.info(f"Downloading {len(gfs_channels)} grib files:")
-    for idname in tqdm(gfs_channels):
-        get_gfs_grib_file(time_gfs, gfs_chunks, idname, f"{GFS_CACHE}/{idname}.grb")
+        # Loop through channels and download grib of each
+        logger.info(f"Downloading {len(gfs_channels)} grib files:")
+        for idname in tqdm(gfs_channels):
+            get_gfs_grib_file(time_gfs, gfs_chunks, idname, f"{GFS_CACHE}/{idname}.grb")
+    else: 
+        logger.info(f"GFS initial conditions already present, skipping download")
```

Comment:
I would put a check in here that the necessary files do exist. If they didn't and someone was looking at the logs, they could get confused as to why the logger says they are present and look in the wrong place for the error. One idea:
```python
    dir_path = pathlib.Path(GFS_CACHE)
    if not dir_path.exists():
        raise FileNotFoundError(f"Directory '{GFS_CACHE}' does not exist.")
    if (do_not_exist := [str(file_path) for idname in gfs_channels if not (file_path := dir_path / f"{idname}.grb").is_file()]):
        raise FileNotFoundError(f"One or more required files missing: {do_not_exist}")
```
...as an aside, I am a huge fan of the walrus operator but don't necessarily approve of lines being this long.

Reply:
Good suggestion, I've added this in in a slightly different way than you suggested but more or less the same (minus the walrus operator).